error/src/internal.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{Error, ErrorKind};
+use failure::{format_err, Error as FailureError, Fail};
+use std::convert::TryFrom;
+use std::fmt::{self, Display};
+
+#[derive(Fail, Debug)]
+pub struct InternalError {
+    kind: InternalErrorKind,
+    cause: Option<String>,
+}
+
+#[derive(Fail, Debug, PartialEq, Eq, Clone, Display)]
+pub enum InternalErrorKind {
+    /// An arithmetic overflow occurs during capacity calculation,
+    /// e.g. `Capacity::safe_add`
+    // NOTE: the original name is {Transaction,Block}::CapacityOverflow
+    ArithmeticOverflowCapacity,
+
+    /// The transaction_pool is already full
+    // NOTE: the original name is LimitReached
+    FullTransactionPool,
+
+    /// Persistent data had corrupted
+    CorruptedData,
+
+    /// Database exception
+    // NOTE: the original name is ckb_db::Error::DBError(String)
+    Database,
+
+    /// VM internal error
+    VM,
+
+    /// The transaction already exist in pool
+    DuplicatedPoolTransaction,
+
```

Comment:
IMO, the subject comes before adjective will be better. such as 
CapacityArithmeticOverflow,
TransactionPoolFull,
PoolTransactionDuplicated...

Reply:
We have discussed in the preview PR, but no conclusion.